nio/responses.py

```diff
@@ -982,6 +984,31 @@ def create_error(parsed_dict, room_id):
         return RoomSendError.from_dict(parsed_dict, room_id)
 
 
+class DirectRoomListErrorResponse(ErrorResponse):
+    pass
+
+
+@dataclass
+class DirectRoomListResponse(Response):
+    """A response containing a list of direct rooms.
+
+    Attributes:
+        rooms (List[str]): The rooms joined by the account.
+    """
+
+    rooms: Dict[str, List[str]] = field()
+
+    @classmethod
+    def from_dict(
+        cls,
+        parsed_dict: Dict[Any, Any],
+    ) -> Union[DirectRoomListResponse, DirectRoomListErrorResponse]:
+        if parsed_dict.get("errcode") == "M_UNRECOGNIZED":
+            # This user has no DM rooms that have been marked with m.direct.
+            return DirectRoomListErrorResponse.from_dict(parsed_dict)
```

Comment:
From testing I only seemed to yield M_UNRECOGNIZED when there were no rooms. Maybe my HS is wrong? I'll try some other servers later

Reply:
I hvae changed this so if there's *any* error code value, the Error response is returned, since under normal conditions that key wouldn't even be there.